lib/ts/ink_endian.h

```diff
@@ -0,0 +1,43 @@
+/** @file
+ *
+ *  Endian convertion routines
+ *
+ *  @section license License
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#pragma once
+
+#ifdef HAVE_SYS_ENDIAN_H
+#include <sys/endian.h>
+#endif
+#ifdef HAVE_MACHINE_ENDIAN_H
+#include <machine/endian.h>
+#endif
+#ifdef HAVE_ENDIAN_H
+#include <endian.h>
+#endif
+#ifdef HAVE_SYS_BYTEORDER_H
+#include <sys/byteorder.h>
+#endif
+
+#if defined(darwin)
+#include <libkern/OSByteOrder.h>
+#define be64toh(x) OSSwapBigToHostInt64(x)
```

Comment:
I think these would be better as inline functions rather than macros.

Reply:
I'm also fine with that, but can you tell me the reasons? To ensure the signatures won't be changed?

Reply:
Primarily because macros don't have name spaces, so every macro name is forbidden in every scope. One of the system header files (cpio.h I think) has a `MAGIC` macro which means I can't have a class static with that name. Inline functions also don't have the extra evaluation issue (e.g. having an expression with side effects). On a philosophical level, IMHO, with C++11 and modern optimization compilers, we should use macros only when absolutely necessary, rather than keeping on with such a crude tool.

Reply:
Well, that makes sense, I think “be64toh” is unique enough and the name rarely conflicts though. I’ll change them to inline functions. Thank you for clarification.